src/components/create-safe/status/useSafeCreation.ts

```diff
@@ -42,7 +42,10 @@ export const addSafeAndOwnersToAddressBook = (pendingSafe: PendingSafeData, chai
     )
 
     pendingSafe.owners.forEach((owner) => {
-      dispatch(upsertAddressBookEntry({ chainId: chainId, address: owner.address, name: owner.name }))
+      const entryName = owner.name || owner.ens
+      if (entryName) {
+        dispatch(upsertAddressBookEntry({ chainId, address: owner.address, name: entryName }))
```

Comment:
I'm not even sure anymore if storing the ens name in the addressbook is such a good idea as we won't double check if the ens name is still associated with that address. So this could lead to some misunderstandings.

We could also offer (cached) ENS resolution in EthHashInfo as an alternative or something similar.

Reply:
I tend to agree but this mirrors the functionality currently found in safe-react.